include/oneapi/dpl/pstl/onedpl_config.h

```diff
@@ -294,4 +310,12 @@
 #    define _ONEDPL_SYCL_INTEL_COMPILER 1
 #endif
 
+// Intel(R) oneAPI DPC++ Compiler cannot build some cases of destroy and destroy_n with unseq policies on Windows
+// It might be fixed in the future releases, which follow 20240000
+#if defined(_MSC_VER) && __INTEL_LLVM_COMPILER <= 20240000
+#    define _ONEDPL_ICPX_OMP_SIMD_DESTROY_WINDWOS_BROKEN 1
+#else
+#    define _ONEDPL_ICPX_OMP_SIMD_DESTROY_WINDWOS_BROKEN 0
+#endif
+
```

Comment:
Do we want to predict that this will be fixed in `__INTEL_LLVM_COMPILER > 20240000`?
This seems like a ticking time bomb.  For me I would rather have just `__INTEL_LLVM_COMPILER` and along with a TODO to reenable it once we have a version which we know works (perhaps a github issue as well to help us remember).

Reply:
Yes, your suggestion looks better. I did it this way so we instantly notice that and raise the version if necessary. It might work well because we have early access to newest versions, but it is definitely not a good idea for pstl_config, which can be unstreamed.

Reply:
I'm also thinking about the case where a user updates the compiler, but not oneDPL (for whatever reason).  
In the case where the compiler bug is one that is not fixed in 2024.1.0, lets say we then update the version in oneDPL for the matching oneAPI release because we see the failing test.  
If the user updates the compiler alone, it will break their code, and moreover look like a *new* compiler bug.  It is unlikely to be clear to them that upgrading to the newest oneDPL would be a resolution for their problem.  

I'd favor being conservative here, but taking some extra precautions (github issue) to see that it isn't forgotten about.